cli/src/etos_client/test_run/test_run.py

```diff
@@ -91,43 +105,50 @@ def track(self, etos: ETOS, timeout: int) -> Events:
                 )
                 break
         self.__wait(etos, end)
-        events = self.__collect(etos)
-        self.__download(events)
-        self.__announce(force=True)
+        self.__download(etos)
+        events = self.__collector.collect(etos.response.tercc)
+        self.__announce(events)
         return events
 
     def __wait(self, etos: ETOS, timeout: int) -> None:
         """Wait for ETOS to finish its activity."""
-        events = self.__collect(etos)
+        events = self.__collector.collect_activity(etos.response.tercc)
         while not events.activity.finished:
             self.logger.info("Waiting for ETOS to finish")
             self.__status(events)
             time.sleep(5)
             if time.time() >= timeout:
                 raise TimeoutError("ETOS did not complete in 24 hours. Exiting")
-            events = self.__collect(etos)
+            events = self.__collector.collect_activity(etos.response.tercc)
 
     def __status(self, events: Events) -> None:
         """Check if ETOS test run has been canceled."""
         if events.activity.canceled:
             raise SystemExit(events.activity.canceled["data"]["reason"])
 
-    def __announce(self, force: bool = False) -> None:
+    def __announce(self, events: Events) -> None:
         """Announce current state of ETOS."""
-        if force or self.__last_log + self.log_interval <= time.time():
-            events = self.__collector.collect_test_case_events()
-            self.__announcer.announce(events, self.__downloader.downloads)
-            self.__last_log = time.time()
-
-    def __download(self, events: Events) -> None:
+        if not events.tercc:
+            self.logger.info("Waiting for ETOS to start.")
+            return
+        if not events.activity.triggered:
+            self.logger.info("Waiting for ETOS to start.")
```

Comment:
Please make the error message different from line 132. It will be easier to understand in the logs where exactly waiting is happening.

Reply:
These messages are for the user, not for us. The difference between a Tercc or activity means nothing to them.
For us, we would see that this check is for either one of these two and could check the ER for more information.

Reply:
@andmat900